squirrel.frontier/src/main/java/org/dice_research/squirrel/queue/scorebasedfilter/UriKeywiseFilter.java

```diff
@@ -0,0 +1,46 @@
+package org.dice_research.squirrel.queue.scorebasedfilter;
+
+import org.dice_research.squirrel.data.uri.CrawleableUri;
+import org.dice_research.squirrel.queue.scorecalculator.IUriScoreCalculator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class filters the {@link CrawleableUri}s to be added to the queue based on the score.
+ */
+public class UriKeywiseFilter implements IUriKeywiseFilter {
```

Comment:
The name is misleading. It is not simply some UriKeywiseFilter. It is mainly based on scores that are calculated for the single URIs. The name should reflect that, e.g., `ScoreBasedUriKeywiseFilter` or something similar. I am sure you can come up with something that fits better :wink: